index/segment/mem/postings_list.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package mem
+
+import (
+	"math"
+	"sync"
+
+	"github.com/m3db/m3ninx/index/segment"
+
+	"github.com/RoaringBitmap/roaring"
+	"github.com/uber-go/atomic"
+)
+
+type postingID int
+
+// postingsList represents an efficient mapping from postingID -> bitmap of doc ids
+type postingsList interface {
+	// InsertNew inserts document `i` at a new posting id, and returns the
+	// positing id used.
+	InsertNew(i segment.DocID) postingID
+
+	// Insert inserts a reference to document `i` for posting id `p`.
+	Insert(p postingID, i segment.DocID)
+
+	// Remove removes the reference (if any) from document `i` from posting id `p`.
+	Remove(p postingID, i segment.DocID)
```

Comment:
This would be hard to support if we want to use delta-encoding? Perhaps we should strictly handle deletes through merges? I also can't imagine we'll often receive deletes for a document soon after it is created. Also, if we don't include `Remove` we can reuse the postings list interface and perhaps implementation between mem and fs.

Reply:
Sold. I'll remove the method